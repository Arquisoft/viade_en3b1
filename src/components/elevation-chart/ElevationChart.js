import React, { Component } from 'react';
import { Line } from 'react-chartjs-2';

export class ElevationChart extends Component {

    constructor(props) {
        super(props);
        this.state = {
            points: props.trackpoints,
        };
        this.init(this.state.points);
    }

    init(points) {
        let index = 1;
        this.data = {
            labels: points.map((p) => `Point ${index++}`),
            datasets: [
                {
                    label: 'Elevation',
                    fill: false,
                    lineTension: 0.1,
                    backgroundColor: 'rgba(75,192,192,0.4)',
                    borderColor: '#44dba1',
                    borderCapStyle: 'butt',
                    borderDash: [],
                    borderDashOffset: 0.0,
                    borderJoinStyle: 'miter',
                    pointBorderColor: '#302c58',
                    pointBackgroundColor: '#302c58',
                    pointBorderWidth: 1,
                    pointHoverRadius: 5,
                    pointHoverBackgroundColor: 'rgba(75,192,192,1)',
                    pointHoverBorderColor: '#302c58',
                    pointHoverBorderWidth: 2,
                    pointRadius: 1,
                    pointHitRadius: 15,
                    data: points.map((p) => parseInt( p.getElevation(), 10 )),
                }
            ]
        };
        this.lineOptions = {
            legend: {
                display: false,
            },
            tooltips: {
                enabled: true,
            },
        };
    }

    render() {
        return (
            <Line
                data={this.data}
                options={this.lineOptions}
                height={100}
            />
        );
    }
}

export default ElevationChart;
