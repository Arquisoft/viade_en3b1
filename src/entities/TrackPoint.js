class TrackPoint {

    /**
     * 
     * @param {Number} latitude 
     * @param {Number} longitude 
     * @param {String} name 
     * @param {Number} elevation 
     */
    constructor(latitude, longitude, name, elevation) {

        this.latitude = latitude;
        this.longitude = longitude;

        if (!name) {
            this.name = "";
        } else {
            this.name = name;
        }

        if(!elevation){
            this.elevation = "";
        } else {
            this.elevation = elevation;
        }
    }

    getName() {
        return this.name;
    }

    getLatitude() {
        return this.latitude;
    }

    getElevation() {
        return this.elevation;
    }

    getLongitude() {
        return this.longitude;
    }

    setElevation(num) {
        this.elevation = num;
    }

    toJson() {
        return {
            "latitude": this.latitude,
            "longitude": this.longitude,
            "elevation": this.elevation,
        };
    }
}

export default TrackPoint;