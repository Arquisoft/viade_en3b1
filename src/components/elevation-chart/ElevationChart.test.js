import ElevationChart from './ElevationChart';
import TrackPoint from '../../entities/TrackPoint';

var point1 = new TrackPoint(1, 2, 3);
var point2 = new TrackPoint(2, 3, 4);
var points = [point1, point2];

test('Define ElevationChart', () => {
  expect(ElevationChart).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = shallow(<ElevationChart trackpoints={points} />);
  expect(wrapper).toMatchSnapshot();
});