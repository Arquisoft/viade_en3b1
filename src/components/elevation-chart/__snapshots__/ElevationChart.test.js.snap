// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<Line
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": "rgba(75,192,192,0.4)",
          "borderCapStyle": "butt",
          "borderColor": "#44dba1",
          "borderDash": Array [],
          "borderDashOffset": 0,
          "borderJoinStyle": "miter",
          "data": Array [
            3,
            4,
          ],
          "fill": false,
          "label": "Elevation",
          "lineTension": 0.1,
          "pointBackgroundColor": "#302c58",
          "pointBorderColor": "#302c58",
          "pointBorderWidth": 1,
          "pointHitRadius": 15,
          "pointHoverBackgroundColor": "rgba(75,192,192,1)",
          "pointHoverBorderColor": "#302c58",
          "pointHoverBorderWidth": 2,
          "pointHoverRadius": 5,
          "pointRadius": 1,
        },
      ],
      "labels": Array [
        "Point 1",
        "Point 2",
      ],
    }
  }
  height={100}
  options={
    Object {
      "legend": Object {
        "display": false,
      },
      "tooltips": Object {
        "enabled": true,
      },
    }
  }
/>
`;
